### PR TITLE
Prove that the initial frame `Ω` is a spectral frame and define its patch

### PR DESCRIPTION
**Pull request by ayberkt**
This PR adds a proof that the initial frame `Ω` is spectral. This is actually quite easy to prove, except there is a technical problem with showing that the basis is closed under binary meets.

Because our definition of spectral locale requires the covering families to be directed, the basis has to be closed under finite joins to ensure that this condition is met. When we do this, however, we end up in a situation where $(x_1 \vee \cdots \vee x_m) \wedge (y_1 \vee \cdots \vee y_n)$ has to be shown to be in the basis, given that each $x_i$ and $y_j$ are basic. This requires a CNF transformation argument that is quite tedious.

One might say that this is a good reason to take out the directedness condition from the definition of spectrality, which I have tried doing in the past. In my experience though, things go much more smoothly if this condition is baked into the definition.

### CODE REVIEW

**Review by martinescardo — APPROVED** [review]

**Comment by ayberkt** [comment]
@martinescardo This is currently a draft PR but I’m in the process of finalising it. Should be ready to review soon.

**Comment by ayberkt** [comment]
Note to self: before proceeding further with this, it‘s probably a good idea to address Issue #168.

File: source/Locales/CompactRegular.lagda [hunk 1]
```diff
@@ -2052,6 +2052,444 @@ module LemmasAboutHeytingComplementation (X : Locale 𝓤 𝓥 𝓥)
...
+
+\begin{code}
+
+directify-preserves-closure-under-∧ : (F : Frame 𝓤 𝓥 𝓦)
+                                    → (ℬ : Fam 𝓦 ⟨ F ⟩)
+                                    → (β : is-basis-for F ℬ)
+                                    → closed-under-binary-meets F ℬ holds
+                                    → let
+                                       ℬ↑ = directify F ℬ
+                                       β↑ = directified-basis-is-basis F ℬ β
```
Comment (martinescardo) [inline source/Locales/CompactRegular.lagda:2389]: This is not used!

File: source/Locales/Sierpinski.lagda [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  h : 𝟚 → ⟨ 𝒪 𝕊 ⟩
+  h ₀ = 𝟎[ 𝒪 𝕊 ]
+  h ₁ = ⊤ₛ
+
+ℬ𝕊-is-basis : is-basis-for (𝒪 𝕊) ℬ𝕊
+ℬ𝕊-is-basis (P , (υ , ι)) =
+ (((P (𝟙 {𝓤} , (λ { ⋆ → ⋆ }) , 𝟙-is-prop)) holds) , λ _ → ₁) , {!!}
+
+ℬ𝕊-is-directed-basis : is-directed-basis (𝒪 𝕊) ℬ𝕊
+ℬ𝕊-is-directed-basis = {!!} , {!!}
```
Comment (martinescardo) [inline source/Locales/Sierpinski.lagda:66]: Incomplete code.

Comment (martinescardo) [inline source/Locales/Sierpinski.lagda:66]: Incomplete code.

File: source/Locales/Sierpinski.lagda [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+𝕊-spectralᴰ : spectralᴰ (𝒪 𝕊)
+𝕊-spectralᴰ = ℬ𝕊 , ℬ𝕊-is-directed-basis , {!!} , {!!}
+
+𝕊-is-spectral : is-spectral (𝒪 𝕊) holds
+𝕊-is-spectral = {!!}
```
Comment (martinescardo) [inline source/Locales/Sierpinski.lagda:72]: Incomplete code.

**Review by martinescardo — APPROVED** [review]

**Review by martinescardo — APPROVED** [review]

**Pull request merged**